src/libs/ajax.js

```diff
@@ -290,6 +290,21 @@ const User = signal => ({
     }
     const res = await fetchBond(`api/link/v1/${provider}/oauthcode?${qs.stringify(queryParams)}`, _.merge(authOpts(), { signal, method: 'POST' }))
     return res.json()
+  },
+
+  isUserRegistered: async email => {
+    try {
+      await fetchSam(`api/users/v1/${email}`, _.merge(authOpts(), { signal, method: 'GET' }))
+    } catch (error) {
+      if (error.status === 404) {
+        return false
+      }
+    }
+    return true
```

Comment:
This will incorrectly return `true` if we catch an error other than 404. We should probably move the `return true` to the end of the `try`, and `throw error` at the end of the `catch`, to make sure other errors get bubbled up and displayed.